src/services/room-service.spec.js

```diff
@@ -1,32 +1,110 @@
     sut = container.get(RoomService);
   });
 
   afterAll(async () => {
     await destroyTestEnvironment(container);
   });
 
+  afterEach(async () => {
+    await pruneTestEnvironment(container);
+  });
+
   describe('createRoom', () => {
     it('should create a room', async () => {
-      const result = await sut.createRoom({ name: 'my room', owner: 'abc', access: ROOM_ACCESS_LEVEL.public });
+      const result = await sut.createRoom({ name: 'my room', access: ROOM_ACCESS_LEVEL.public, user: myUser });
 
       expect(result).toEqual({
         _id: expect.stringMatching(/\w+/),
         name: 'my room',
-        owner: 'abc',
+        owner: myUser._id,
         access: ROOM_ACCESS_LEVEL.public,
         members: []
       });
     });
   });
+
+  describe('findOwnedRoomById', () => {
+    let myRoom = null;
+    let otherRoom = null;
+
+    beforeEach(async () => {
+      [myRoom, otherRoom] = await Promise.all([
+        sut.createRoom({ name: 'my room', access: ROOM_ACCESS_LEVEL.public, user: myUser }),
+        sut.createRoom({ name: 'not my room', access: ROOM_ACCESS_LEVEL.public, user: otherUser })
+      ]);
+    });
+
+    it('should find rooms owned by the specified user ID', async () => {
+      const room = await sut.findOwnedRoomById({ roomId: myRoom._id, ownerId: myUser._id });
+      expect(room.name).toBe('my room');
+    });
+
+    it('should not find rooms owned by other users', async () => {
+      const room = await sut.findOwnedRoomById({ roomId: otherRoom._id, ownerId: myUser._id });
+      expect(room).toBeNull();
+    });
+  });
+
+  describe('createOrUpdateInvitation', () => {
+    let myPublicRoom = null;
+    let myPrivateRoom = null;
+
+    beforeEach(async () => {
+      [myPublicRoom, myPrivateRoom] = await Promise.all([
+        await sut.createRoom({ name: 'my public room', access: ROOM_ACCESS_LEVEL.public, user: myUser }),
+        await sut.createRoom({ name: 'my private room', access: ROOM_ACCESS_LEVEL.private, user: myUser })
+      ]);
+    });
+
+    it('should create a new invitation if it does not exist', async () => {
+      const { invitation } = await sut.createOrUpdateInvitation({ roomId: myPrivateRoom._id, email: 'invited-user@test.com', user: myUser });
+      expect(invitation.token).toBeDefined();
+    });
+
+    it('should update an invitation if it already exists', async () => {
+      const { invitation: originalInvitation } = await sut.createOrUpdateInvitation({ roomId: myPrivateRoom._id, email: 'invited-user@test.com', user: myUser });
+      const { invitation: updatedInvitation } = await sut.createOrUpdateInvitation({ roomId: myPrivateRoom._id, email: 'invited-user@test.com', user: myUser });
+      expect(updatedInvitation._id).toBe(originalInvitation._id);
+      expect(updatedInvitation.token).not.toBe(originalInvitation.token);
+      expect(updatedInvitation.sentOn).not.toBe(originalInvitation.sentOn);
+      expect(updatedInvitation.expires).not.toBe(originalInvitation.expires);
```

Comment:
The assertion could be that the new expires needs to be higher than the old one. But I'm not insisting.

Reply:
Good point, I'll do that!

Reply:
Done